core/data/src/test/java/uk/gov/gchq/gaffer/data/element/function/ReduceRelatedElementsTest.java

```diff
@@ -0,0 +1,453 @@
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex2a)
+                        .dest(vertex3b)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("private"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex2b)
+                        .dest(vertex3a)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex3a)
+                        .dest(vertex4b)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("private"))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex5b)
+                        .dest(vertex4a)
+                        .directed(true)
+                        .group(RELATES_TO)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .build()
+        );
+
+        // When
+        final Iterable<Element> result = function.apply(elements);
+
+        // Then
+        final List<Element> expectedElements = Arrays.asList(
+                new Edge.Builder()
+                        .source(vertex1a)
+                        .dest(vertex2b)
+                        .directed(true)
+                        .group(TestGroups.EDGE)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .property("sourceRelatedVertices", Sets.newHashSet(vertex1b))
+                        .property("destinationRelatedVertices", Sets.newHashSet(vertex2a))
+                        .build(),
+                new Edge.Builder()
+                        .source(vertex1a)
+                        .dest(vertex3b)
+                        .directed(true)
+                        .group(TestGroups.EDGE)
+                        .property(VISIBILITY, Sets.newHashSet("public", "private"))
+                        .property("sourceRelatedVertices", Sets.newHashSet(vertex1b))
+                        .property("destinationRelatedVertices", Sets.newHashSet(vertex3a))
+                        .build(),
+                new Entity.Builder()
+                        .vertex(vertex2b)
+                        .group(TestGroups.ENTITY)
+                        .property(VISIBILITY, Sets.newHashSet("public"))
+                        .property("relatedVertices", Sets.newHashSet(vertex2a))
+                        .build()
+        );
+//        assertElementEquals(expectedElements, result);
+    }
+
+    @Test
+    public void shouldReturnInputIfNoRelatedVertexGroups() {
+        Object vertex1a = "vertex1a longest";
```

Comment:
Might be worth making these constants at the class level if they're reused in multiple tests